projects/Mallard/src/components/layout/ui/with-breakpoints.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState, ReactElement } from 'react'
+import { View, LayoutRectangle, StyleSheet } from 'react-native'
+
+const WithBreakpoints = ({
+    children,
+}: {
+    children: {
+        0: (l: LayoutRectangle) => ReactElement
+        [fromSize: number]: (l: LayoutRectangle) => ReactElement
+    }
+}) => {
+    const [maxSize, setMaxSize] = useState<keyof typeof children>(0)
+    const [metrics, setMetrics] = useState<LayoutRectangle | null>(null)
+    return (
+        <View
+            onLayout={ev => {
```

Comment:
How come we're checking the width of the children here rather than just the parent `View` itself?